File: worldedit-core/src/main/java/com/boydti/fawe/beta/implementation/processors/NullProcessor.java
package com.boydti.fawe.beta.implementation.processors;

import com.boydti.fawe.beta.IBatchProcessor;
import com.boydti.fawe.beta.IChunk;
import com.boydti.fawe.beta.IChunkGet;
import com.boydti.fawe.beta.IChunkSet;
import com.sk89q.worldedit.extent.Extent;
import com.sk89q.worldedit.extent.NullExtent;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.concurrent.Future;

public final class NullProcessor implements IBatchProcessor {
    private static final NullProcessor instance = new NullProcessor();

    public static NullProcessor getInstance() {
        return instance;
    }

    @Nullable
    public IChunkSet processSet(@NotNull IChunk chunk, @NotNull IChunkGet get, @NotNull IChunkSet set) {
        return null;
    }

    @Nullable
    public Future<IChunkSet> postProcessSet(@NotNull IChunk chunk, @NotNull IChunkGet get, @NotNull IChunkSet set) {
        return null;
    }

    @NotNull
    public Extent construct(@NotNull Extent child) {
        return new NullExtent();
    }

    private NullProcessor() {
    }

}

File: worldedit-core/src/main/java/com/boydti/fawe/beta/implementation/processors/NullProcessor.kt
package com.boydti.fawe.beta.implementation.processors

import com.boydti.fawe.beta.IBatchProcessor
import com.boydti.fawe.beta.IChunk
import com.boydti.fawe.beta.IChunkGet
import com.boydti.fawe.beta.IChunkSet
import com.sk89q.worldedit.extent.Extent
import com.sk89q.worldedit.extent.NullExtent

object NullProcessor : IBatchProcessor {

    override fun processSet(chunk: IChunk, get: IChunkGet, set: IChunkSet): IChunkSet? {
        return null
    }

    override fun construct(child: Extent): Extent {
        return NullExtent()
    }
}
